### ISSUE TITLE
Make indented code blocks an extension

### ISSUE THREAD

**Issue by jtojnar** [jgm/pandoc#3751]
I never use indented code blocks. On the other hand, sometimes I need to embed a slightly more complex HTML structure (e.g. a table) into a document. Indentation would make the HTML code clearer; unfortunately, it turns the table into a code listing.

While it is [possible](https://stackoverflow.com/questions/39220389/embed-indented-html-in-markdown-with-pandoc) to have indented HTML, AFAIK it cannot be done without the loss of `markdown_in_html_blocks` which I would like to keep. For that reason, it would be nice if the indented code blocks could be disabled.

**Comment by jgm** [comment]
Would #3537 help here?  Then you could write
````
```{=html}
<tag>

     <tag>
         indent as much as you want
     </tag>

</tag>
```
````

**Comment by jgm** [comment]
I've had this request before, but I'm somewhat reluctant to make core Markdown features (as opposed to extensions) selectable.

**Comment by jtojnar** [comment]
I guess that #3537 would be somewhat better than the current solution since `markdown_in_html_blocks` would still be enabled globally. I would still prefer if I could use markdown link syntax inside the table, though. 

Maybe it could be possible to require an empty line before indented code blocks like Commonmark [does](http://spec.commonmark.org/0.27/#indented-code-block)?

**Comment by jgm** [comment]
Not true that Commonmark requires a blank line before indented code.  It stipulates that indented code  cannot interrupt a paragraph.  But it can come directly after another kind of construction, e.g. [in this case](http://spec.commonmark.org/dingus/?text=---%0A%20%20%20%20hi%0A%0A).

**Comment by tarleb** [comment]
I believe this can be considered solved by 82cc7fb0d462401b54bfe5172e7e49ab7b7302d9.

**Comment by beedell-roke** [comment]
@tarleb, that doesn't make it an extension. For those who want to actually disable it, does that help?
